mitodl/open-discussions
# Use react-helmet-async instead of react-meta-tags, add noindex meta tag to 404/403 error pages

#### Pre-Flight checklist



#### What are the relevant tickets?
Closes #3752 

#### What's this PR do?
- Adds a meta robots (noindex,noarchive) tag on 404 results to open-discussions pages
- Replaces `react-meta-tags` with `react-helmet-async`
- Adds a `MetaTags` component that renders head tags via `react-helmet-async`
- Moves the `ErrorPage` components from the `open-discussions` frontend to `ol-util` so it can be used elsewhere

#### How should this be manually tested?
Go to the profile page of any non-existent user, ex http://localhost:8063/profile/fakeuser - view the source via the Elements tab in Chrome developer tools, you should see this at the end:

```
<meta name="robots" content="noindex,noarchive" data-rh="true">
```

Go to some other modified pages in open-discussions using `MetaTags` to set title, for example http://localhost:8063/login/ - make sure the title is set correctly.


#### Where should the reviewer start?
Lots of the changed files are just swapping the `react-meta-tags` component for the new `MetaTags` component wrapper for `<Helmet>`.  The most relevant changes are here:

frontends/infinite-corridor/package.json
frontends/infinite-corridor/src/App.tsx

frontends/ol-util/src/components/ErrorPage.tsx
frontends/ol-util/src/components/MetaTags.tsx
frontends/ol-util/src/components/index.ts

frontends/open-discussions/package.json
frontends/open-discussions/src/Router.js
frontends/open-discussions/src/util/integration_test_helper.js

#### Any background context you want to provide?
`react-helmet-async`'s <Helmet> component doesn't seem to allow custom components as children, so the `CanonicalLink` component was removed,  and a `canonicalUrl` prop was added to the <MetaTags> component wrapper around <Helmet>, which only adds a <link> child if not null/empty.

The infinite-corridor frontend still needs to be modified to return the noindex metatag on 404's, I'll do that in a separate PR.
